drivers/counter/counter_nxp_pit.c

```diff
@@ -0,0 +1,407 @@
+	PIT_Type *base;
+	bool enableRunInDebug;
+	int num_channels;
+#if DT_NODE_HAS_PROP(DT_COMPAT_GET_ANY_STATUS_OKAY(nxp_pit), interrupts)
+	void (*irq_config_func)(const struct device *dev);
+#else
+	void (**irq_config_func)(const struct device *dev);
+#endif
+	const struct device *clock_dev;
+	clock_control_subsys_t clock_subsys;
+	struct nxp_pit_channel_data *const *data;
+	const struct device *const *channels;
+};
+
+static uint32_t nxp_pit_get_top_value(const struct device *dev)
+{
+	const struct nxp_pit_config *config = dev->config;
+	pit_chnl_t channel = PIT_CHANNEL_ID(dev);
+
+	/*
+	 * According to RM, the LDVAL trigger = clock ticks -1
+	 * The underlying HAL driver function PIT_SetTimerPeriod()
+	 * automatically subtracted 1 from the value that ends up in
+	 * LDVAL so for reporting purposes we need to add it back in
+	 * here to by consistent.
+	 */
+	return (config->base->CHANNEL[channel].LDVAL + 1);
+}
+
+static int nxp_pit_start(const struct device *dev)
+{
+	const struct nxp_pit_config *config = dev->config;
+	int channel_id = PIT_CHANNEL_ID(dev);
+
+	LOG_DBG("period is %d", nxp_pit_get_top_value(dev));
+	PIT_EnableInterrupts(config->base, channel_id,
+			     kPIT_TimerInterruptEnable);
+	PIT_StartTimer(config->base, channel_id);
+	return 0;
+}
+
+static int nxp_pit_stop(const struct device *dev)
+{
+	const struct nxp_pit_config *config = dev->config;
+	int channel_id = PIT_CHANNEL_ID(dev);
+
+	PIT_DisableInterrupts(config->base, channel_id,
+			      kPIT_TimerInterruptEnable);
+	PIT_StopTimer(config->base, channel_id);
+
+	return 0;
+}
+
+static int nxp_pit_get_value(const struct device *dev, uint32_t *ticks)
+{
+	const struct nxp_pit_config *config = dev->config;
+	int channel_id = PIT_CHANNEL_ID(dev);
+
+	*ticks = PIT_GetCurrentTimerCount(config->base, channel_id);
+
+	return 0;
+}
+
+static int nxp_pit_set_top_value(const struct device *dev,
+				  const struct counter_top_cfg *cfg)
+{
+	const struct nxp_pit_config *config = dev->config;
+	struct nxp_pit_channel_data *data = PIT_CHANNEL_DATA(dev);
+	pit_chnl_t channel = PIT_CHANNEL_ID(dev);
+
+	if (cfg->ticks == 0) {
+		return -EINVAL;
+	}
+
+	data->top_callback = cfg->callback;
+	data->top_user_data = cfg->user_data;
+
+	if (config->base->CHANNEL[channel].TCTRL & PIT_TCTRL_TEN_MASK) {
+		/* Timer already enabled, check flags before resetting */
+		if (cfg->flags & COUNTER_TOP_CFG_DONT_RESET) {
+			return -ENOTSUP;
+		}
+		PIT_StopTimer(config->base, channel);
+		PIT_SetTimerPeriod(config->base, channel, cfg->ticks);
+		PIT_StartTimer(config->base, channel);
+	} else {
+		PIT_SetTimerPeriod(config->base, channel, cfg->ticks);
+	}
+
+	return 0;
+}
+
+static uint32_t nxp_pit_get_pending_int(const struct device *dev)
+{
+	const struct nxp_pit_config *config = dev->config;
+	uint32_t mask = PIT_TFLG_TIF_MASK;
+	uint32_t flags;
+	int channel_id = PIT_CHANNEL_ID(dev);
+
+	flags = PIT_GetStatusFlags(config->base, channel_id);
+
+	return ((flags & mask) == mask);
```

Comment:
should return 1 or 0 not true or false

Reply:
? Those are equivalent statements, are they not? 
Even with regards to the way they are defined by the API
I am sure the one who documented the API understood
the returning statement 1 as "true, there are pending interrupts"
and 0 as "false, there are no pending interrupts"

Reply:
I guess I was taught and it was drilled into me that the relational operators and true are not guaranteed to be 1... but on some research to the C standard, it seems maybe this teaching was wrong or the C standard changed in later revisions... C11 says == will yieldand int 1 if true, so maybe it's fine then